java/code/src/com/suse/manager/reactor/messaging/RegisterMinionEventMessageAction.java

```diff
@@ -944,7 +944,8 @@ private String getOsRelease(String minionId, ValueMap grains) {
                 "oel".equalsIgnoreCase(grains.getValueAsString("os")) ||
                 "alibaba cloud (aliyun)".equalsIgnoreCase(grains.getValueAsString("os")) ||
                 "almalinux".equalsIgnoreCase(grains.getValueAsString("os")) ||
-                "amazon".equalsIgnoreCase(grains.getValueAsString("os")) ||
+                ("amazon".equalsIgnoreCase(grains.getValueAsString("os")) &&
+                 "2".equalsIgnoreCase(grains.getValueAsString("osmajorrelease"))) ||
```

Comment:
On 2023 `/etc/redhat-release` does not exist, and I guess it will not exist in newer versions either.

Reply:
But this parse grains. So it might use "/etc/os-release" and this should exist

Reply:
Might still me correct, but what product is created when you register AZ2023 ?

Reply:
Without this, the problem is that line 951 looks for the file `redhat-release` which does not exist in Amazon Linux 2023.

What you mean is that I should be doing an if starting at line 950 instead so then line 958 ` osRelease = whatprovidesRes.flatMap(res -> res.fold(` is still processed?

Reply:
I think as long as we do not add sumatoolbox support we might not need this.
Is there any product shown in the UI when you register this OS?
I think it is ok, if we check again in case we want to add official SUMA support for it.

Reply:
![imagen](https://github.com/uyuni-project/uyuni/assets/4226070/371d6729-b175-40fd-bfc9-8b4941424aa6)

Unknown, as it's not in sumatoolbox/SCC yet.

As discussed in Slack, this  can be revisited if that situation arrives. For now this was needed as otherwise the onboarding fails because the java code looks  for the `redhat-release` file, which doesn't exist.